File: Back/NodeJS/src/domain/friendships/Friendship.ts
import { Entity } from "../Entity";
import { MemberId } from "../members/MemberId";
import { FriendshipsRequested } from './events/FriendshipsRequested';

export class Friendship extends Entity {

    private constructor(requesterId: MemberId, memberId: MemberId){
        super()
        this.FriendId = memberId;
        this.RequesterId = requesterId;
        this.AddDomainEvent(new FriendshipsRequested(requesterId.Value, memberId.Value));
    }

    static Create(requesterId: MemberId, memberId: MemberId) {
        return new Friendship(requesterId, memberId);
    }
    public FriendId: MemberId;
    public RequesterId: MemberId;
}

File: Back/.Net/BetFriends.Domain/Friendships/Friendship.cs
using BetFriends.Domain.Events;
using BetFriends.Shared.Domain;

namespace BetFriends.Domain.Friendships;

public class Friendship : Entity
{
    private readonly Guid friendId;
    private readonly Guid requesterId;

    private Friendship(Guid friendId, Guid requesterId)
    {
        this.friendId = friendId;
        this.requesterId = requesterId;
        AddEvent(new FriendshipRequested(requesterId, friendId));
    }

    public Guid FriendId { get => friendId; }
    public Guid RequesterId { get => requesterId; }

    internal static Friendship Create(Guid memberId, Guid requesterId)
    {
        return new Friendship(memberId, requesterId);
    }
}
